MyLibrary/Sources/MyLibrary/WeatherService.swift

```diff
@@ -4,15 +4,21 @@ public protocol WeatherService {
     func getTemperature() async throws -> Int
 }
 
-class WeatherServiceImpl: WeatherService {
-    let url = "https://api.openweathermap.org/data/2.5/weather?q=corvallis&units=imperial&appid=<INSERT YOUR API KEY HERE>"
+enum ServiceDomain : String {
+  case actual = "https://api.openweathermap.org"
+  case mock = "http://localhost:3000"
+}
 
+class WeatherServiceImpl: WeatherService {
+    let url = "\(ServiceDomain.mock.rawValue)/data/2.5/weather?q=corvallis&units=imperial&appid=INSERT_YOUR_API_KEY_HERE"
```

Comment:
While you are at it, why not have a constant called `apiKey` that you can string-interpolate here?  You are already using _string interpolation_ to insert the base url.